packages/code-studio/src/styleguide/ThemeColors.tsx

```diff
@@ -16,16 +16,29 @@ import {
   INVALID_COLOR_BORDER_STYLE,
 } from './colorUtils';
 
-const swatchDataGroups = {
-  'Theme Color Palette': buildColorGroups('palette', palette, 1),
-  'Semantic Colors': buildColorGroups('semantic', semantic, 1),
-  'Chart Colors': buildColorGroups('chart', chart, 2),
-  'Editor Colors': buildColorGroups('editor', semanticEditor, 2),
-  'Grid Colors': buildColorGroups('grid', semanticGrid, 2),
-  'Component Colors': buildColorGroups('component', components, 1),
-};
+function buildSwatchDataGroups() {
+  return {
+    'Theme Color Palette': buildColorGroups('palette', palette, 1),
+    'Semantic Colors': buildColorGroups('semantic', semantic, 1),
+    'Chart Colors': buildColorGroups('chart', chart, 2),
+    'Editor Colors': buildColorGroups('editor', semanticEditor, 2),
+    'Grid Colors': buildColorGroups('grid', semanticGrid, 2),
+    'Component Colors': buildColorGroups('component', components, 1),
+  };
+}
 
 export function ThemeColors(): JSX.Element {
+  const [swatchDataGroups, setSwatchDataGroups] = useState(
+    buildSwatchDataGroups
+  );
+
+  const { selectedThemeKey } = useTheme();
+
+  useEffect(() => {
+    // Rebuild swatches when themes change
+    setSwatchDataGroups(buildSwatchDataGroups());
+  }, [selectedThemeKey]);
```

Comment:
I think you can just use `useMemo` here.
```suggestion
  const swatchDataGroups = useMemo(buildSwatchDataGroups, [selectedThemeKey]);
```

Reply:
Cool. eslint complained about unused selectedThemeKey when I tried:

```
useMemo(() => buildSwatchDataGroups(), [selectedThemeKey])
```

Looks like your version doesn't have this issue, presumably because eslint can't see inside the closure